src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -448,23 +439,14 @@ else if (tokenType == TokenTypes.LITERAL_CATCH) {
         private static Details getDetailsForIfElse(DetailAST ast) {
             boolean shouldCheckLastRcurly = false;
             final DetailAST lcurly;
-            DetailAST nextToken;
-            final int tokenType = ast.getType();
-            if (tokenType == TokenTypes.LITERAL_IF) {
```

Comment:
Else and if are handled and computed exactly the same thanks to the already existing `null` check.